public/app/core/services/backend_srv.ts

```diff
@@ -212,6 +213,14 @@ export class BackendSrv implements BackendService {
               const firstAttempt = i === 0 && options.retry === 0;
 
               if (error.status === 401 && isLocalUrl(options.url) && firstAttempt && isSignedIn) {
+                if (error.data.message.includes('revoked')) {
+                  appEvents.emit(CoreEvents.showModalReact, {
+                    component: TokenRevokedModal,
+                  });
+
+                  return of(null);
```

Comment:
If we really need this if logic, I think this should be a separate function and we should add RxJs tests for that function.